docs/sources/reference/api/docker_remote_api_v1.19.md

```diff
@@ -1125,6 +1127,143 @@ Status Codes:
 -   **500** – server error
 
+### Get an archive of a filesytem resource in a container
+
+`GET /containers/(id)/archive-path`
+
+Get an Tar archive of a resource in the filesystem of container `id`.
+
+Query Parameters:
+
+- **path** - resource in the container's filesystem to archive. Required.
+
+    If not an absolute path, it is relative to the container's root directory.
+    The resource specified by **path** must exist. To assert that the resource
+    is expected to be a directory, **path** should end in `/` or  `/.` (asuming
+    a path separator of `/`). If **path** ends in `/.` then this indicates that
+    only the contents of the **path** directory should be copied. A symlink
+    followed by a `/` is resolved to its target.
+
+    > It is not possible to copy certain system files such as resources under
+    > `/proc`, `/sys`, `/dev`, and mounts created by the user in the container.
+
+**Example request**:
+
+        GET /containers/8cce319429b2/archive-path?path=/foo/ HTTP/1.1
+
+**Example response**:
+
+        HTTP/1.1 200 OK
+        Content-Type: application/x-tar
+
+        {{ TAR STREAM }}
+
+Status Codes:
+
+- **200** - success, returns archive of copied resource
+- **400** - client error, bad parameter, details in JSON response body, one of:
+    - must specify path parameter (**path** cannot be empty)
+    - not a directory (**path** was asserted to be a directory but exists as a
+      file)
+- **404** - client error, resource not found, one of:
+    – no such container (container `id` does not exist)
+    - no such file or directory (**path** does not exist)
+- **500** - server error
+
+### Extract an archive of files or folders to a directory in a container
+
+`PUT /containers/(id)/extract-to-dir`
+
+Upload a Tar archive to be extracted to a path in the filesystem of container
+`id`.
+
+Query Parameters:
+
+- **path** - path to a directory in the container
+    to extract the archive's contents into. Required.
+
+    If not an absolute path, it is relative to the container's root directory.
+    The **path** resource must exist.
+
+**Example request**:
+
+        PUT /containers/8cce319429b2/extract-to-dir?path=/vol1 HTTP/1.1
+        Content-Type: application/x-tar
+
+        {{ TAR STREAM }}
+
+**Example response**:
+
+        HTTP/1.1 200 OK
+
+Status Codes:
+
+- **200** – the content was extracted successfully
+- **400** - client error, bad parameter, details in JSON response body, one of:
+    - must specify path parameter (**path** cannot be empty)
+    - not a directory (**path** should be a directory but exists as a file)
+- **403** - client error, permission denied, the volume
+    or container rootfs is marked as read-only.
+- **404** - client error, resource not found, one of:
+    – no such container (container `id` does not exist)
+    - no such file or directory (**path** resource does not exist)
+- **500** – server error
+
+### Stat a file or directory in a container
+
+`GET /containers/(id)/stat-path`
```

Comment:
The more I think about this the more I would really prefer a more RESTful solution.  Not going to -1 the PR over it, but adding a Docker-specific endpoint to do what, normally, an HTTP HEAD should do feels like a mistake.
Even if this PR doesn't use the "GET a single file" endpoint, I think having that endpoint so that this PR can use HEAD against it to retrieve the metadata about the file in question I think would be a better (more generic) solution and less tied to a specific cli command.

@crosbymichael any thoughts on this?